File: backend/cinema-backend/src/main/java/com/example/gestion_cinema/service/TicketService.java
package com.example.gestion_cinema.service;

import com.example.gestion_cinema.entity.Ticket;

import java.util.List;

public interface TicketService {

    Ticket add(Ticket reservation);
    List<Ticket> findAll();
    Ticket save(Ticket reservation);
    List<Ticket> findByUserId(Long id);
    void delete(Long id);


}

File: frontend/src/app/service/ticket.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { ticket } from '../common/ticket';
import { environment } from '../environments/environment';

@Injectable({
  providedIn: 'root'
})
export class TicketService {

  constructor(private _http: HttpClient) { }

  baseUrl = environment.baseUrl;

  findAll() {
    return this._http.get<ticket[]>(this.baseUrl + "ticket/find-all");
  }

  delete(id: any) {
    return this._http.delete(this.baseUrl + `ticket/${id}`);
  }
  findByUserId(id: any) {
    return this._http.get<ticket[]>(this.baseUrl + `ticket/find-by-user-id/${id}`);
  }
  
  annulerRes(id:any){
    return this._http.delete(this.baseUrl + `ticket/${id}`);
  }

  save(ticket: any) {
    return this._http.post(this.baseUrl + "ticket/add", ticket)
  }
}